core/services/workflows/monitoring.go

```diff
@@ -0,0 +1,85 @@
+package workflows
+
+import (
+	"context"
+	"fmt"
+	"github.com/smartcontractkit/chainlink-common/pkg/beholder"
+	"github.com/smartcontractkit/chainlink-common/pkg/beholder/pb"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"google.golang.org/protobuf/proto"
+	"log"
+)
+
+var registerTriggerFailureCounter metric.Int64Counter
+
+func initMonitoringResources(_ context.Context) (err error) {
+	registerTriggerFailureCounter, err = beholder.GetMeter().Int64Counter("RegisterTriggerFailure")
+	if err != nil {
+		return fmt.Errorf("failed to register trigger failure: %s", err)
+	}
+	return nil
+}
+
+// TODO: tradeoff between having labels be variadic vs values be variadic?
+func sendLogAsCustomMessageWithLabels(ctx context.Context, labels map[string]string, format string, values ...interface{}) (err error) {
+	for labelKey, labelValue := range labels {
+		ctx, err = KeystoneContextWithLabel(ctx, labelKey, labelValue)
+		if err != nil {
+			return err
+		}
+	}
+
+	return sendLogAsCustomMessage(ctx, format, values...)
```

Comment:
Hmmmm... so IIUC this is adding some labels to the context only to fetch them out of the context down below?

That feels clunky to me: we should either add the labels inside the engine whenever we have a label to add, or just pass in labels explicitly (probably the first option is preferred)

It would be fine by the way to also extract those labels for logging, since AFAIU they need to be the same anyway